odl/deform/linear_deformation.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Shape-based reconstruction using optimal information transportation.
+
+The Fisher-Rao metric is used in regularization term. And L2 data matching
+term is used in fitting term.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+
+from odl.operator import Operator
+from odl.discr import DiscreteLp, Divergence
+from odl.space import ProductSpace
+
+__all__ = ('LinearDeformation', 'MassPreservingLinearDeformation')
+
+
+class LinearDeformation(Operator):
+    """Linearized deformation."""
+    def __init__(self, offsets=None):
```

Comment:
> Well the middle one requires some knowledge of a reasonable "basis" for the deformation, i.e. some vector space structure or similar. RKHS would be fine, but that should (i.m.o.) be added in another branch by someone more competent in that than I am.

I think the middle one can be implemented without any knowledge of RKHS whatsoever. Just take a vector field (could lie in an RKHS) and apply it to the given template via the action of linearized deformation.

Agree in principle, but since we know that we are going to need at least the first two cases (actually so far we *always* considered the middle one), we should create structures for them already now. Going forward with an arbitrary choice and breaking it later seems a bit pointless, we already have that bit of information.

> My suggestion here is that we write this class with a nice interface and an acceptable implementation and merge this.
> 
> Then once we add more deformations, we have more hands on information on what we'll need so we can start factoring out things like that.